compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -396,17 +399,35 @@ private void defineTypeNodes(List<BLangTypeDefinition> typeDefs, SymbolEnv env)
     }
 
     private void checkErrors(BLangTypeDefinition unresolvedType, BLangType currentTypeNode, List<String> visitedNodes,
-            List<LocationData> encounteredUnknownTypes) {
+                             List<LocationData> encounteredUnknownTypes) {
         String unresolvedTypeNodeName = unresolvedType.name.value;
-        // Type node can be either user defined type or union type. Finite types will not be added to the
-        // unresolved list because they can be resolved.
-        if (currentTypeNode.getKind() == NodeKind.UNION_TYPE_NODE) {
+
+        // Check errors in the type definition.
+        if (currentTypeNode.getKind() == NodeKind.ARRAY_TYPE) {
+            checkErrors(unresolvedType, ((BLangArrayType) currentTypeNode).elemtype, visitedNodes,
+                    encounteredUnknownTypes);
+        } else if (currentTypeNode.getKind() == NodeKind.UNION_TYPE_NODE) {
             // If the current type node is a union type node, we need to check all member nodes.
             List<BLangType> memberTypeNodes = ((BLangUnionTypeNode) currentTypeNode).memberTypeNodes;
             for (BLangType memberTypeNode : memberTypeNodes) {
-                if (memberTypeNode.getKind() == NodeKind.UNION_TYPE_NODE) {
+                if (memberTypeNode.getKind() == NodeKind.ARRAY_TYPE) {
```

Comment:
For each member, shouldn't we call the same method itself?

Code block within this for-loop seems duplicate.